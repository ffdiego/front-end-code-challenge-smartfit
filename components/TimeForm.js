export default function TimeForm({ search, results }) {
  function handleForm(event) {
    event.preventDefault();
    const form = event.target;
    if (form.address.value.length > 1) {
      search(form);
    }
  }

  return (
    <div className="my-4 border-4 rounded border-sf-light-grey p-4 font-gotham text-sf-dark-grey">
      <p className="font-gotham mb-3">
        <img src="/images/icon-hour.png" className="inline h-6 mr-2" />
        Horário
      </p>
      <p className="border-b-2 text-lg mb-3">Qual período quer treinar?</p>
      <form onSubmit={handleForm}>
        <table className="table-auto w-full">
          <tbody>
            <tr className="border-b">
              <td>
                <input
                  type="radio"
                  id="manha"
                  name="period"
                  value="manha"
                  className="mr-2 my-3"
                />
                <label htmlFor="manha">Manhã</label>
              </td>
              <td className="text-right">06:00 às 12:00</td>
            </tr>
            <tr className="border-b">
              <td>
                <input
                  type="radio"
                  id="tarde"
                  name="period"
                  value="tarde"
                  className="mr-2 my-3"
                />
                <label htmlFor="tarde">Tarde</label>
              </td>
              <td className="text-right">12:01 às 18:00</td>
            </tr>
            <tr className="border-b">
              <td>
                <input
                  type="radio"
                  id="noite"
                  name="period"
                  value="noite"
                  className="mr-2 my-3"
                />
                <label htmlFor="noite">Noite</label>
              </td>
              <td className="text-right">18:01 às 23:00</td>
            </tr>
          </tbody>
        </table>

        <input
          className="w-full border-2 rounded my-2 p-4 outline-none focus:border-sf-yellow transition duration-300"
          placeholder="Insira aqui um endereço..."
          name="address"
        />
        <div className="flex flex-col sm:flex-row justify-between">
          <span>
            <input
              type="checkbox"
              id="showClosed"
              name="showClosed"
              className="mr-2 mt-3"
            />
            <label htmlFor="showClosed">Exibir unidades fechadas</label>
          </span>
          <p>
            Resultados encontrados: <b>{results.length || 0}</b>
          </p>
        </div>
        <div className="flex justify-evenly">
          <button className="w-64 border-2 border-sf-yellow bg-sf-yellow text-black font-gotham font-bold py-2 px-4 rounded mt-4">
            ENCONTRAR UNIDADE
          </button>
          <input
            value="LIMPAR"
            type="reset"
            className="w-64 border-2 border-sf-light-grey text-black font-gotham font-bold py-2 px-4 rounded mt-4 cursor-pointer"
          />
        </div>
      </form>
    </div>
  );
}
