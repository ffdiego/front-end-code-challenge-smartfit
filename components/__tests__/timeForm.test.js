import TimeForm from "components/TimeForm";

import { render } from "@testing-library/react";
import UserEvent from "@testing-library/user-event";

describe("Time Form tests", () => {
  it("should render correctly", () => {
    const { container } = render(<TimeForm results={[]} />);
    expect(container).toMatchSnapshot();
  });

  it("should not call search function when there is no input", () => {
    const search = jest.fn();
    const { getByText } = render(<TimeForm search={search} results={[]} />);
    const button = getByText("ENCONTRAR UNIDADE");
    UserEvent.click(button);
    expect(search).not.toHaveBeenCalled();
  });

  it("should call search function when there is input", () => {
    const search = jest.fn();
    const { getByText } = render(<TimeForm search={search} results={[]} />);
    const radio = getByText("Noite");
    const button = getByText("ENCONTRAR UNIDADE");

    UserEvent.click(radio);
    UserEvent.click(button);

    expect(search).toHaveBeenCalled();
  });
});
