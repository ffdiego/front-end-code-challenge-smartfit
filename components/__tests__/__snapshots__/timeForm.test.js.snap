// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Form tests should render correctly 1`] = `
<div>
  <div
    class="my-4 border-4 rounded border-sf-light-grey p-4 font-gotham text-sf-dark-grey"
  >
    <p
      class="font-gotham mb-3"
    >
      <img
        class="inline h-6 mr-2"
        src="/images/icon-hour.png"
      />
      Horário
    </p>
    <p
      class="border-b-2 text-lg mb-3"
    >
      Qual período quer treinar?
    </p>
    <form>
      <table
        class="table-auto w-full"
      >
        <tbody>
          <tr
            class="border-b"
          >
            <td>
              <input
                class="mr-2 my-3"
                id="manha"
                name="period"
                type="radio"
                value="manha"
              />
              <label
                for="manha"
              >
                Manhã
              </label>
            </td>
            <td
              class="text-right"
            >
              06:00 às 12:00
            </td>
          </tr>
          <tr
            class="border-b"
          >
            <td>
              <input
                class="mr-2 my-3"
                id="tarde"
                name="period"
                type="radio"
                value="tarde"
              />
              <label
                for="tarde"
              >
                Tarde
              </label>
            </td>
            <td
              class="text-right"
            >
              12:01 às 18:00
            </td>
          </tr>
          <tr
            class="border-b"
          >
            <td>
              <input
                class="mr-2 my-3"
                id="noite"
                name="period"
                type="radio"
                value="noite"
              />
              <label
                for="noite"
              >
                Noite
              </label>
            </td>
            <td
              class="text-right"
            >
              18:01 às 23:00
            </td>
          </tr>
        </tbody>
      </table>
      <input
        class="w-full border-2 rounded my-2 p-4 outline-none focus:border-sf-yellow transition duration-300"
        hidden=""
        name="address"
        placeholder="Insira aqui um endereço..."
      />
      <div
        class="flex flex-col sm:flex-row justify-between mt-6"
      >
        <span>
          <input
            class="mr-2 mt-3"
            id="showClosed"
            name="showClosed"
            type="checkbox"
          />
          <label
            for="showClosed"
          >
            Exibir unidades fechadas
          </label>
        </span>
        <p>
          Resultados encontrados: 
          <b>
            0
          </b>
        </p>
      </div>
      <div
        class="flex justify-evenly"
      >
        <button
          class="w-64 border-2 border-sf-yellow bg-sf-yellow text-black font-gotham font-bold py-2 px-4 rounded mt-4"
        >
          ENCONTRAR UNIDADE
        </button>
        <input
          class="w-64 border-2 border-sf-light-grey text-black font-gotham font-bold py-2 px-4 rounded mt-4 cursor-pointer"
          type="reset"
          value="LIMPAR"
        />
      </div>
    </form>
  </div>
</div>
`;
